storage/storage.go

```diff
@@ -41,3 +41,19 @@ type STHWriter interface {
 type RootsWriter interface {
 	WriteRoots(ctx context.Context, l *ctlog.Log, roots []*x509.Certificate, receivedAt time.Time) error
```

Comment:
I was letting the RootsWriter implementation handle calculating the RootSetID, since it's arguably implementation-specific. The RootSetID isn't defined by Monologue to be anything but an opaque, deterministic identifier. I could make WriteRoots() return the RootSetID, which makes this a bit more obvious?